raiden/waiting.py

```diff
@@ -499,7 +502,7 @@ def wait_for_received_transfer_result(
                 result = TransferWaitResult.UNLOCK_FAILED
                 break
 
-        state_changes = raiden.wal.storage.get_state_changes()
+        state_changes = next(stream)
```

Comment:
Isn't this a change in behaviour? I looks like a single element instead of the whole list.

Reply:
No, it's not. I just got confused by the `Iterator[List[StateChange]]`, expecting `Iterator[StateChange]`.